compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilationOptions.java

```diff
@@ -40,13 +40,15 @@ public class CompilationOptions {
     Boolean enableCache;
     Boolean disableSyntaxTree;
     Boolean remoteManagement;
-
-    CompilationOptions(Boolean offlineBuild, Boolean observabilityIncluded, Boolean dumpBir,
-                       Boolean dumpBirFile, String cloud, Boolean listConflictedClasses, Boolean sticky,
-                       Boolean dumpGraph, Boolean dumpRawGraphs, Boolean withCodeGenerators,
-                       Boolean withCodeModifiers, Boolean configSchemaGen, Boolean exportOpenAPI,
-                       Boolean exportComponentModel, Boolean enableCache, Boolean disableSyntaxTree,
-                       Boolean remoteManagement) {
+    static Boolean optimizeCodegen;
+    static Boolean verbose;
```

Comment:
Other variables in that class were also following the same pattern. I will talk with @gayaldassanayake and clarify the reason for it.

Reply:
They are using Boolean wrapper to check whether the field was initialized or not. And if it is not initialized the compiler will return false. [This method](https://github.com/ballerina-platform/ballerina-lang/blob/fcab497029f10a9424fa7c26f9592248d57843ee/compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilationOptions.java#L228) cannot be implemented if we use the base type.

Reply:
After a discussion with @azinneera it turned out the actual reason for the boxed type usage is because, we have to keep track of three states(true/false/not-provided). The initial value for the boxed type will be `null` and it can be used to represent the `not-provided` state. 

This information is useful for deciding the final flag value when there are multiple sources for the flag values. For example,
```
Ballerina.toml - true, CLI - false=> Final value is false
Ballerina.toml - true, CLI - not-provided => Final value is true
Ballerina.toml - true, CLI - false => Final value is false
```
The overriding cannot be done without the `null` state for the flag.